src/vm/luca_loaderwriter.cpp

```diff
@@ -0,0 +1,75 @@
+#include "common.h"
+#include "luca_loaderwriter.h"
+#include "pal.h"
+
+LoaderWriter::LoaderWriter(int pid) {
+    LIMITED_METHOD_CONTRACT;
+
+    this->pid = pid;
+
+    SString path;
+    path.Printf("loader-%d.map", pid);
+    this->OpenFile(path);
+}
+
+void LoaderWriter::WriteRecord(SString& command, SString& path, SString& guid) {
```

Comment:
I would recommend that you turn this into a method that just knows how to write a line (e.g. a type and a value), and then create another method that builds the line out of the information that you want to store.